README.md

```diff
@@ -353,33 +415,73 @@ We have to create a constructor in the `Human` class that takes a `Dog` instance
 as an argument and assigns it to the `dog` field in the class. Then we annotate
 the constructor with `@Autowired`.
 
+The most common way to use the `@Autowired` annotation is to create a constructor
+in the class that will reference another bean. In our case, we'll create an
+autowired constructor in the `Human` class that takes a `Dog` instance as an
+argument.
+
 ```java
 @Component
 public class Human {
     private String name;
-    private final Dog dog;
+    private Dog dog;
 
     @Autowired
     public Human(Dog dog) {
-        this.dog = dog;
+       
     }
+    
+    // getters and setters
+}
+```
+
+Since we are using the `@Autowired` annotation, we don't have to instantiate the
+`dog` field in the `Human` class - we can just assign it to the
+`Dog` argument passed into the constructor:
 
-		// getters, setters, and toString
-		// no setter for "dog" since it's a final field
+```java
+@Component
+public class Human {
+    private String name;
+    private Dog dog;
+
+    @Autowired
+    public Human(Dog dog) {
+       this.dog = dog;
+    }
+    
+    // getters and setters
+}
+```
+
+We can now also make the `dog` instance variable `final` as well, since the value
+is injected during object creation.
+
+```java
+@Component
+public class Human {
+    private String name;
+    private final Dog dog;
+
+    @Autowired
+    public Human(Dog dog) {
+       this.dog = dog;
+    }
+    
+    // getters and setters
+    // no setter for the dog since it is now a final field
 }
 ```
 
-If we run the `main` method we will get the following output:
+If we run the application we will get the following output:
```

Comment:
I'm getting an error when it attempts to get the human bean 		
Human lily = context.getBean(Human.class);

Exception in thread "main" org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'com.example.demo.domain.Human' available
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.getBean(DefaultListableBeanFactory.java:351)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.getBean(DefaultListableBeanFactory.java:342)
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1172)
	at com.example.demo.DemoApplication.main(DemoApplication.java:17)

Reply:
Ok that works.!

You might want to include the current output, since it has all null values for the state:

Human{name='null', dog=Dog{name='null'}}
Dog{name='null'}

Reply:
Ok that works.!

You might want to include the current output, since it has all null values for the state. Not sure if that is what you want, since it differs from previous example:

Human{name='null', dog=Dog{name='null'}}
Dog{name='null'}

Reply:
Ok that works.!

You might want to include the current output, since it has all null values for the state. Not sure if that is what you want, since it differs from previous example:

Human{name='null', dog=Dog{name='null'}}
Dog{name='null'}